homeassistant/components/media_player/cambridgeaudio.py

```diff
@@ -0,0 +1,382 @@
+        except OSError:
+            _LOGGER.debug("Unable to connect to device at %s", host)
+
+    elif discovery_info is not None:    # netdisco found the device
+        addr = discovery_info.get('host')
+        port = discovery_info.get('port')
+
+        # if a name was set in the config, use it
+        if name is None:
+            name = discovery_info.get('name')
+
+        scpd_url = discovery_info.get('ssdp_description')
+        if addr not in KNOWN_HOSTS:
+            smdevice = cadevice.StreamMagicDevice(addr, port, name, scpd_url)
+            hosts.append(CADevice(smdevice, config.get(CONF_SOURCES),
+                                  poweroff_command, name=name))
+            KNOWN_HOSTS.append(host)
+            _LOGGER.debug("Added StreamMagic device with ip %s (%s)",
+                          addr, name)
+    add_devices(hosts, True)
+
+
+class CADevice(MediaPlayerDevice):
+    """Representation of a Cambridge Audio Network Audio Player device."""
+
+    def __init__(self, smdevice, sources, poweroff_command, name=None):
+        """Initialization."""
+        self._smdevice = smdevice
+        self._name = name
+        self._pwstate = STATE_UNKNOWN
+        self._state = STATE_UNKNOWN
+        self._muted = False
+        self._support_volumecontrol = self._smdevice.get_volume_control()
+        self._volume = 0
+        self._volume_max = self._smdevice.get_volume_max()
+        self._source = None
+        self._sources_map = dict((pr_num, pr_name)
+                                 for pr_num, pr_name, pr_state
+                                 in self._smdevice.get_preset_list())
+        self._sources_reverse = {name: id for id, name
+                                 in self._sources_map.items()}
+        self._source_list = list(self._sources_map.values())
+        self._audio_source = None
+        self._power_off_cmd = poweroff_command.upper()  # IDLE or OFF
+        self._artist = None
+        self._album_art = None
+        self._album = None
+        self._trackno = None
+        self._title = None
+        self._duration = None
+        self._position = None
+        self._position_updated_at = None
+
+    @property
+    def name(self):
+        """Return the name of the media_player."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the media_player."""
+        return self._state
+
+    @property
+    def is_volume_muted(self):
+        """Return the state of the muting function."""
+        return self._muted
+
+    @property
+    def source(self):
+        """Return currently selected input source."""
+        return self._audio_source
+
+    @property
+    def source_list(self):
+        """Return a list of available Presets."""
+        return self._source_list
+
+    @property
+    def supported_features(self):
+        """Return supported features."""
+        features = SUPPORT_CAMBRIDGE_COMMON
+        if self._audio_source == "media player":
```

Comment:
Supported features should not change during the lifetime of the entity. This property doesn't seem to honor that. I'm presuming audio source can be changed by the user by selection in the source list.